File: Modules/SMES/MES/SIE.Web.MES/SpcFromMess/Commands/ReadonlyViewSpcFromMesCommand.cs
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.SpcFromMess.Commands
{
    /// <summary>
    ///  查看预览命令（仅前端逻辑）
    /// </summary>
    [JsCommand("SIE.Web.MES.SpcFromMess.Commands.ReadonlyViewSpcFromMesCommand")]
    public class ReadonlyViewSpcFromMesCommand : ViewCommand
    {
        /// <summary>
        /// Excute
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return true;
        }
    }
}

File: Modules/SMES/MES/SIE.Web.MES/SpcFromMess/Commands/ReadonlyViewSpcFromMesCommand.js
SIE.defineCommand('SIE.Web.MES.SpcFromMess.Commands.ReadonlyViewSpcFromMesCommand', {
    extend: 'SIE.cmd.Edit',
    meta: { text: "数据预览", group: "edit", iconCls: "icon-PageSearch icon-blue" },

    canExecute: function (view) {
        if (!view) { return false; }
        if (view.getSelection && typeof view.getSelection === 'function') {
            return view.getSelection().length === 1;
        }
        return true;
    },

    execute: function (view, source) {
        var selection = view.getSelection();
        if (selection && selection.length === 1) {
            this.showView(selection[0]);
        } else {
            SIE.Msg.showWarning('请选择一条记录');
        }
    },

    showView: function (editEntity) {
        if (!editEntity) { return; }
        var me = this;
        CRT.Workbench.addPage({
            tabId: 'spc_readonly_' + String(editEntity.getId()).replace(/[.|,]/g, ''),
            entityType: me.view.model,
            recordId: editEntity.getId(),
            viewGroup: "ReadonlyView",
            title: Ext.String.format('数据预览-{0}'.L10N(), editEntity.getNo()),
            isDetail: true,
            params: {
                token: me.view.token,
                action: 3
            }
        });
    }
});
